src/server/services/tool-registration.service.ts

```
@@ -0,0 +1,271 @@
+      },
+      signal: new AbortController().signal,
+      requestId: randomUUID(),
+    };
+  }
+
+  /**
+   * Resolve client project root from arguments or stored map
+   */
+  private resolveClientProjectRoot(args: ToolArguments): string | undefined {
+    let effectiveClientProjectRoot = (args as any).clientProjectRoot;
+    
```

ESLint

[error] Delete `····`

(prettier/prettier)